### PR TITLE
Dev 9988 support buildah

### PR DESCRIPTION
**Pull request by TomerHasson**

### CODE REVIEW

File: pkg/controlplane/tmpl/conf/rbac/buildimage-job-role.tpl [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  namespace: {{ ns . }}
+  annotations:
+    {{- range $k, $v := .Spec.Annotations }}
+    {{$k}}: "{{$v}}"
+    {{- end }}
+  labels:
+    {{- range $k, $v := .Spec.Labels }}
+    {{$k}}: "{{$v}}"
+    {{- end }}
+{{- if not .Spec.ControlPlane.BaseConfig.CnvrgJobRbacStrict }}
```
Comment (Dimss) [inline pkg/controlplane/tmpl/conf/rbac/buildimage-job-role.tpl:14]: remove rules, we don't need it here

File: pkg/controlplane/tmpl/conf/cm/config-base.tpl [hunk 2]
```diff
@@ -35,6 +35,8 @@ data:
   HAS_CNVRG_SCHEDULER: "{{ .Spec.ControlPlane.CnvrgScheduler.Enabled }}"
   CNVRG_SSO_REALM: "{{ .Spec.SSO.RealmName }}"
   CNVRG_SSO_SERVICE_URL: "{{ .Spec.SSO.ServiceUrl }}"
+  CRI: "{{ .Spec.Cri }}"
```
Comment (Dimss) [inline pkg/controlplane/tmpl/conf/cm/config-base.tpl:38]: Rename `CRI` to `CNVRG_CRI`

**Review by Dimss — CHANGES_REQUESTED** [review]

File: pkg/controlplane/tmpl/conf/rbac/buildimage-job-role.tpl (hunk 1 above)
Comment (TomerHasson) [reply pkg/controlplane/tmpl/conf/rbac/buildimage-job-role.tpl:14]: fixed

File: controllers/utils.go [hunk 3]
```diff
@@ -56,7 +59,29 @@ func generateSecureToken(length int) string {
 	return hex.EncodeToString(b)
 }
 
-func calculateAndApplyAppDefaults(app *mlopsv1.CnvrgApp, desiredAppSpec *mlopsv1.CnvrgAppSpec, infra *mlopsv1.CnvrgInfra) {
+func DiscoverCri(clientset client.Client) (mlopsv1.CriType, error) {
+	nodeList := &v1.NodeList{}
+	if err := clientset.List(context.Background(), nodeList, client.Limit(1)); err != nil {
+		return "", err
```
Comment (Dimss) [inline controllers/utils.go:65]: handle error correctly -> require reconcile on error

File: controllers/utils.go [hunk 4]
```diff
@@ -56,7 +59,29 @@ func generateSecureToken(length int) string {
...
+	}
+	node := nodeList.Items[0]
+	cri := node.Status.NodeInfo.ContainerRuntimeVersion
+	if strings.Contains(cri, string(mlopsv1.CriTypeContainerd)) {
```
Comment (Dimss) [inline controllers/utils.go:69]: we need explicitly support for all cris 
if we are not able to detect the cri -> fall into the error state 
supported cris are:
docker
containerd
cri-o

**Review by Dimss — CHANGES_REQUESTED** [review]

File: controllers/cnvrginfra_controller.go [hunk 5]
```diff
@@ -295,6 +304,17 @@ func (r *CnvrgInfraReconciler) applyManifests(cnvrgInfra *mlopsv1.CnvrgInfra) er
 	return reconcileResult
 }
 
+func getCriClass(criType mlopsv1.CriType) string {
+	switch criType {
+	case mlopsv1.CriTypeContainerd:
+		return "cri"
+	case mlopsv1.CriTypeDocker:
+		return "docker"
+	default:
```
Comment (Dimss) [inline controllers/cnvrginfra_controller.go]: remove the default

File: pkg/logging/tmpl/fluentbit/cm.tpl [hunk 6]
```diff
@@ -29,14 +29,15 @@ data:
     @INCLUDE {{ $app.SpecName }}-{{ $app.SpecNs }}-output.conf
     {{- end }}
 
+  {{ $criType := .Data.CriType }}
   {{- range $_, $app := .Data.AppInstance }}
 
   {{ $app.SpecName }}-{{ $app.SpecNs }}-input.conf: |
```
Comment (Dimss) [inline pkg/logging/tmpl/fluentbit/cm.tpl]: use global scope `$.`

**Review by Dimss — APPROVED** [review]
looks good

**Review by Dimss — DISMISSED** [review]

**Review by Dimss — DISMISSED** [review]
look good

**Review by Dimss — APPROVED** [review]

**Pull request merged**